packages/core/src/core/plan/plan.ts

```
@@ -160,6 +162,50 @@ const isEqualsNode = async (
   return _.isEqual(node1, node2)
 }
 
+type BeforeAfterElement = {
+  before: ChangeDataType | undefined
+  after: ChangeDataType | undefined
+}
+
+export async function *iterateTogether(before: AsyncIterable<Element>,
+  after: AsyncIterable<Element>): AsyncIterable<BeforeAfterElement> {
+  const beforeIterator = before[Symbol.asyncIterator]()
+  const afterIterator = after[Symbol.asyncIterator]()
+  const nextDefinedValue = async (
+    iter: AsyncIterator<Element | undefined>
+  ): Promise<ChangeDataType | undefined> => {
+    const next = await iter.next()
+    if (next.done) {
+      return undefined
+    }
+    return next.value ? next.value as ChangeDataType : nextDefinedValue(iter)
+  }
+  let currentBefore = await nextDefinedValue(beforeIterator)
+  let currentAfter = await nextDefinedValue(afterIterator)
+  while (!(currentBefore === undefined && currentAfter === undefined)) {
+    if (currentBefore === undefined) {
+      yield { before: undefined, after: currentAfter }
+      // eslint-disable-next-line no-await-in-loop
+      currentAfter = await nextDefinedValue(afterIterator)
+    } else if (currentAfter === undefined
+      || currentBefore.elemID.getFullName() < currentAfter.elemID.getFullName()) {
```

I'd probably split into two ifs for readbility

---

Also - should probably throw an error if the iterator keys are not sorted ( just throw an error of an iterator key has lower value then the previous key obtained by it )